examples/convert_osi2mcap.cpp

```diff
@@ -0,0 +1,205 @@
+    {"TrafficCommand", osi3::ReaderTopLevelMessage::kTrafficCommand},  {"TrafficCommandUpdate", osi3::ReaderTopLevelMessage::kTrafficCommandUpdate},
+    {"TrafficUpdate", osi3::ReaderTopLevelMessage::kTrafficUpdate},    {"MotionRequest", osi3::ReaderTopLevelMessage::kMotionRequest},
+    {"StreamingUpdate", osi3::ReaderTopLevelMessage::kStreamingUpdate}};
+
+void printHelp() {
+    std::cout << "Usage: convert_osi2mcap <input_file> <output_file> [--input-type <message_type>]\n\n"
+              << "Arguments:\n"
+              << "  input_file              Path to the input OSI trace file\n"
+              << "  output_file             Path to the output MCAP file\n"
+              << "  --input-type <message_type>   Optional: Specify input message type if not stated in filename\n\n"
+              << "Valid message types:\n";
+    for (const auto& [type, _] : kValidTypes) {
+        std::cout << "  " << type << "\n";
+    }
+}
+
+std::optional<ProgramOptions> parseArgs(const int argc, const char** argv) {
+    if (argc < 3 || std::string(argv[1]) == "--help" || std::string(argv[1]) == "-h") {
+        printHelp();
+        return std::nullopt;
+    }
+
+    ProgramOptions options;
+    options.input_file_path = argv[1];
+    options.output_file_path = argv[2];
+    options.message_type = osi3::ReaderTopLevelMessage::kUnknown;
+
+    for (int i = 3; i < argc; i++) {
+        if (std::string arg = argv[i]; arg == "--input-type" && i + 1 < argc) {
+            const std::string type_str = argv[++i];
+            auto types_it = kValidTypes.find(type_str);
+            if (types_it == kValidTypes.end()) {
+                std::cerr << "Error: Invalid message type '" << type_str << "'\n\n";
+                printHelp();
+                return std::nullopt;
+            }
+            options.message_type = types_it->second;
+        }
+    }
+
+    return options;
+}
+
+int main(const int argc, const char** argv) {
+    const auto options = parseArgs(argc, argv);
+    if (!options) {
+        return 1;
+    }
+
+    std::cout << "Input file: " << options->input_file_path << std::endl;
+    std::cout << "Output file: " << options->output_file_path << std::endl;
+
+    auto trace_file_reader = osi3::SingleChannelBinaryTraceFileReader();
+    if (!trace_file_reader.Open(options->input_file_path, options->message_type)) {
+        std::cerr << "ERROR: Could not open input file " << options->input_file_path << std::endl;
+        return 1;
+    }
+
+    auto trace_file_writer = osi3::MCAPTraceFileWriter();
+    if (!trace_file_writer.Open(options->output_file_path)) {
+        std::cerr << "ERROR: Could not open output file " << options->output_file_path << std::endl;
+        return 1;
+    }
+
+    // according to the OSI specification mcap must contain a timestamp and zero_time metadata entry
+    // try to parse the file timestamp from the .osi file name (as it should follow the recommended OSI naming conventions)
```

Comment:
What happens if the naming convention is not used in the .osi trace file?

Reply:
Or do we require this here, because we need the message type anyways?

Reply:
See https://github.com/Lichtblick-Suite/asam-osi-utilities/pull/2#discussion_r1865433822

Now the `zero_time` is simply not set if not present in the filename.

```
$ ./convert_osi2mcap /tmp/test.osi /tmp/test_converted.mcap --input-type SensorView
Input file: "/tmp/test.osi"
Output file: "/tmp/test_converted.mcap"
Finished single channel binary to mcap converter

$ mcap list metadata /tmp/test_converted.mcap
name              	offset	length	metadata                                                                                                                                                                                                                                  	
net.asam.osi.trace	46    	282   	{"creation_time":"2024-12-02T08:37:58.1Z","description":"Converted from /tmp/test.osi","max_osi_version":"3.7.0","max_protobuf_version":"3.21.12","min_osi_version":"3.7.0","min_protobuf_version":"3.21.12","version":"3.7.0"}	
```

Reply:
Proposal: Provide type e.g. "sv" by parameter. Throw error if not present in name and no parameter is set. Throw warning if name indicates other type then parameter. parameter overwrites name type.

Reply:
> Proposal: Provide type e.g. "sv" by parameter. Throw error if not present in name and no parameter is set. Throw warning if name indicates other type then parameter. parameter overwrites name type.

That is already implemented 

```bash
$ ./convert_osi2mcap --help
Usage: convert_osi2mcap <input_file> <output_file> [--input-type <message_type>]

Arguments:
  input_file              Path to the input OSI trace file
  output_file             Path to the output MCAP file
  --input-type <message_type>   Optional: Specify input message type if not stated in filename

Valid message types:
  MotionRequest
  TrafficCommandUpdate
  TrafficCommand
  HostVehicleData
  SensorView
  TrafficUpdate
  SensorData
  StreamingUpdate
  GroundTruth
  
$ ./convert_osi2mcap /tmp/test.osi /tmp/test_converted.mcap
Input file: "/tmp/test.osi"
Output file: "/tmp/test_converted.mcap"
ERROR: Unable to determine message type from the filename '"/tmp/test.osi"'. Please ensure the filename follows the recommended OSI naming conventions as specified in the documentation or specify the message type manually.
ERROR: Could not open input file "/tmp/test.osi

```

Reply:
> Throw warning if name indicates other type then parameter. parameter overwrites name type.

Done in https://github.com/Lichtblick-Suite/asam-osi-utilities/pull/2/commits/52215386d69ea4d186e2497ac52fff79fdede3b5 and https://github.com/Lichtblick-Suite/asam-osi-utilities/pull/2/commits/0df13a4432946a5a08f13f386e62d56c5f642e0f